Gems/ScriptAutomation/Code/Source/ImGui/ImGuiAssetBrowser.cpp

```diff
@@ -0,0 +1,468 @@
+        }
+    }
+
+    ImGuiAssetBrowser::ImGuiAssetBrowser(AZStd::string_view configFilePath)
+    {
+        m_configFilePath = configFilePath;
+    }
+
+    void ImGuiAssetBrowser::Activate()
+    {
+        // The original m_configFilePath passed to the constructor likely starts with "@user@" which needs to be replaced with the real path.
+        // The constructor is too early to resolve the path so we do it here on activation.
+        char configFileFullPath[AZ_MAX_PATH_LEN] = {0};
+        AZ::IO::FileIOBase::GetInstance()->ResolvePath(m_configFilePath.c_str(), configFileFullPath, AZ_MAX_PATH_LEN);
+        m_configFilePath = configFileFullPath;
+
+        AzFramework::AssetCatalogEventBus::Handler::BusConnect();
+    }
+
+    void ImGuiAssetBrowser::Deactivate()
+    {
+        if (!m_configFilePath.empty())
+        {
+            // We only report this message in Deactivate(), not inside SaveConfigFile(), to avoid spamming
+            // this message when SaveConfigFile() is called in OnTick()
+            AZ_TracePrintf("ImGuiAssetBrowser", "Saved settings to '%s'\n", m_configFilePath.c_str());
+
+            SaveConfigFile();
+        }
+
+        AzFramework::AssetCatalogEventBus::Handler::BusDisconnect();
+    }
+
+    void ImGuiAssetBrowser::OnCatalogAssetAdded(const AZ::Data::AssetId& assetId)
+    {
+        OnCatalogChanged(assetId);
+    }
+        
+    void ImGuiAssetBrowser::OnCatalogAssetRemoved(const AZ::Data::AssetId& assetId, const AZ::Data::AssetInfo&)
+    {
+        OnCatalogChanged(assetId);
+    }
+
+    void ImGuiAssetBrowser::OnCatalogChanged(const AZ::Data::AssetId& assetId)
+    {
+        AZ::Data::AssetInfo assetInfo;
+        AZ::Data::AssetCatalogRequestBus::BroadcastResult(assetInfo, &AZ::Data::AssetCatalogRequestBus::Events::GetAssetInfoById, assetId);
+
+        if (m_includedAssetFilter && m_includedAssetFilter(assetInfo))
+        {
+            m_needsRefresh = true;
+        }
+    }
+
+    void ImGuiAssetBrowser::SetFilter(AssetFilterCallback shouldInclude)
+    {
+        m_includedAssetFilter = shouldInclude;
+    }
+
+    void ImGuiAssetBrowser::PopulateAssets(AZStd::function<bool(const AZ::Data::AssetInfo& assetInfo)> shouldInclude)
+    {
+        m_assets.clear();
+        m_pinnedAssets.clear();
+        m_configFile.m_pinnedAssetPaths.clear();
+        m_prevSelectedAssetIndex = -1;
+        m_selectedAssetIndex = -1;
+        m_selectedPinnedAssetIndex = -1;
+
+        auto startCB = []() {};
+
+        auto enumerateCB = [this,shouldInclude](const AZ::Data::AssetId id, const AZ::Data::AssetInfo& assetInfo)
+        {
+            if (shouldInclude(assetInfo))
+            {
+                Utils::AssetEntry entry;
+                entry.m_path = assetInfo.m_relativePath;
+                entry.m_assetId = id;
+                entry.m_name = assetInfo.m_relativePath;
+
+                m_assets.push_back(entry);
+            }
+        };
+
+        auto endCB = []() {};
```

Comment:
Does this function need to save the supplied shouldInclude function as the includedAssetFilter?